civihr_employee_portal/js/reports.js

```diff
@@ -919,6 +919,67 @@
                         }
 
                         this.$DOM.sections.dataWrapper.fadeIn();
+                        var $DOM = this.$DOM;
+
+                        $.ajax({
```

Comment:
I believe the jQuery version we are using is already using promises for ajax calls, so you can write something like this:

```js
$.ajax({
    type: 'GET',
    url: buildURL.call(this),
})
.then(function (data) {
    this.$DOM.sections.dataWrapper.html(data).ready(/* ... */);
})
.then(function () {
    return $ajax({
        'type': 'GET',
        'url': this.getJsonUrl() + ( typeof path !== 'undefined' ? path : '' )
    });
})
.then(function (data) {
    var mappedResults = {}

    /* .... */
});